index.js

```diff
@@ -205,6 +205,18 @@ function sendRequest(client, device, message, projectId, accessToken, doneCallba
             // Convert response body to JSON object
             let response = JSON.parse(data);
 
+            // Extract status code from JSON response object
+            let statusCode = response.statusCode ?? response.status;
```

Comment:
One more thing - can you please move the entire code before `let response = JSON.parse(data);`? Google likes to return HTML when it returns `500 Internal Server Error` and that will cause the `JSON.parse()` to throw before your code gets executed.